### DISCUSSION TITLE
Allow urls for X-Sendfile header

### DISCUSSION THREAD

**Discussion by rmikalkenas** [roadrunner-server/roadrunner#1842] [Ideas]
At the moment X-Sendfile header only allows local files.
What if it would also allow to pass an url to remote file which would be streamed?
An example could be file streaming from S3 bucket:
backend generates files bucket url and passes it as X-Sendfile header value.
Although not sure about possible security implications, would be interesting to discuss.

**Comment by rustatian** [comment]
Hey @rmikalkenas 👋 

Hm, I'm also not sure about passing an S3 presigned URL via headers... But since they are passed only locally, so we should be sure that RR deletes that header after processing in any situation. Generally, In Go we have a 'panic handler' for this, so, if something goes wrong (panicking), we would be able to clear headers with the sensitive information.

**Reply by rmikalkenas** [reply]
I will try to prepare a proof of concept :+1:

**Reply by rmikalkenas** [reply]
@rustatian finally managed to find some time for this > https://github.com/roadrunner-server/send/pull/99
